bin/copy-labels.js

```diff
@@ -0,0 +1,285 @@
+
+const help = `\
+This tool copies labels from source repository or json file to destination
+repository or json file.
+
+Usage: ${toolName} [OPTION]... SOURCE DEST
+       ${toolName} --help
+       ${toolName} --version
+
+Options:
+  --config  path to the config file
+  --user    GitHub username
+  --token   GitHub access token
+  --delete  delete unused labels in destination repository before copy
+  --update  update existing labels with the same name
+  --dry-run print operations that will be executed and exit
+  --help    print this help message and exit
+  --version print version and exit
+`;
+
+const options = new Map([
+  ['--config', 'config'],
+  ['--user', 'user'],
+  ['--token', 'token'],
+  ['--delete', 'delete'],
+  ['--update', 'update'],
+  ['--dry-run', 'dry'],
+]);
+
+const helpExit = (message, error) => {
+  console.error(message);
+  if (error) console.error(error.message);
+  console.error(`\nTry '${toolName} --help' for more information`);
+  process.exit(1);
+};
+
+const handleArgs = () => {
+  const args = { args: [] };
+  for (const arg of process.argv.slice(2)) {
+    const [opt, value] = arg.split('=');
+    if (opt.startsWith('--help')) {
+      console.log(help);
+      process.exit(0);
+    } else if (opt.startsWith('--version')) {
+      console.log(toolVersion);
+      process.exit(0);
+    } else if (options.has(opt)) {
+      args[options.get(opt)] = value || true;
+    } else if (!opt.startsWith('--') && args.args.length < 2) {
+      args.args.push(opt);
+    } else {
+      helpExit('Unrecognized option:' + arg);
+    }
+  }
+  return args;
+};
+
+const printLabels = labels => {
+  console.table(
+    labels.reduce((acc, label) => {
+      acc[label.name] = label;
+      return acc;
+    }, {}),
+    ['color', 'description']
+  );
+};
+
+const readDump = async file => {
+  const readFile = util.promisify(fs.readFile);
+  let data;
+  try {
+    data = JSON.parse(await readFile(file));
+  } catch (err) {
+    helpExit(`Error reading json file: ${file}`, err);
+  }
+  return data;
+};
+
+const writeDump = async (file, data) => {
+  const result = data.map(label => ({
+    name: label.name,
+    color: label.color,
+    description: label.description || undefined,
+  }));
+
+  const writeFile = util.promisify(fs.writeFile);
+  try {
+    await writeFile(file, JSON.stringify(result, null, 2));
+  } catch (err) {
+    helpExit(`Error writing file: ${file}`, err);
+  }
+};
+
+const dryDeleteLabels = async (args, handler) => {
+  const labels = await handler.get();
+  const usedLabels = await handler.getUsedLabels(args.dstRepo);
+  console.log(`Delete following labels from '${args.dstRepo}'`);
+  printLabels(labels.filter(({ name }) => !usedLabels.has(name)));
+  return usedLabels;
+};
+
+const printAddedLabels = (args, labels, existingLabels, usedLabels) => {
+  console.log('Following labels will be added:');
+  printLabels(
+    labels.filter(label => {
+      for (const existingLabel of existingLabels) {
```

Comment:
Why not make `existingLabels` into a `Map` to avoid this loop every time here and below?

Reply:
But then we will need to convert them back to the array to filter here
https://github.com/metarhia/tools/blob/289c28cffda940a0c708617e4e99f685351ee84d/bin/copy-labels.js#L143
and here
https://github.com/metarhia/tools/blob/289c28cffda940a0c708617e4e99f685351ee84d/bin/copy-labels.js#L157
If anything I'd like to just create `Set` where we need this.